react/src/Sidebar/NestedSidebarItem.tsx

```diff
@@ -0,0 +1,58 @@
+import { FC, PropsWithChildren, useMemo } from 'react'
+import {
+  Accordion,
+  AccordionButton,
+  AccordionIcon,
+  AccordionItem,
+  AccordionPanel,
+  AccordionProps,
+  Flex,
+  Text,
+  VStack,
+} from '@chakra-ui/react'
+import { merge } from 'lodash'
+
+import { useSidebarStyles } from './Sidebar'
+import { SidebarItemType } from './types'
+
+export interface NestedSidebarItemProps extends SidebarItemType {
+  /**
+   * Whether the item is the root item of the sidebar. Determines styling.
+   */
+  root?: boolean
+  containerProps?: AccordionProps
```

Comment:
wdyt about making it also extend `AccordionProps` (or use a `type` and intersections) so that users can treat it like an accordion also, which seems inline with the usage